monitors.go

```diff
@@ -96,61 +81,42 @@ func (m *Monitor) JobUpdate(updateKey aurora.JobUpdateKey, interval int, timeout
 				return false, nil
 			}
 		}
-
-		fmt.Println("Polling, update still active...")
-		time.Sleep(time.Duration(interval) * time.Second)
+		return false, nil
+	})
+	if err != nil {
+		return false, err
 	}
-
-	fmt.Println("Timed out")
-	return false, nil
+	return true, nil
 }
 
 func (m *Monitor) Instances(key *aurora.JobKey, instances int32, interval int, timeout int) (bool, error) {
 
 	defaultBackoff := m.Client.RealisConfig().backoff
-	duration := defaultBackoff.Duration
 	var err error
 	var live map[int32]bool
 
-	for i := 0; i*interval < timeout; i++ {
-		for step := 0; step < defaultBackoff.Steps; step++ {
-			if step != 0 {
-				adjusted := duration
-				if defaultBackoff.Jitter > 0.0 {
-					adjusted = Jitter(duration, defaultBackoff.Jitter)
-				}
-				fmt.Println(" sleeping for: ", adjusted)
-				time.Sleep(adjusted)
-				fmt.Println(" sleeping done")
-				duration = time.Duration(float64(duration) * defaultBackoff.Factor)
-			}
-			if live, err = m.Client.GetInstanceIds(key, aurora.LIVE_STATES); err == nil {
-				fmt.Println(" live: ", live)
-				break
-			}
-
-			if err != nil {
-				err1 := m.Client.ReestablishConn()
-				if err1 != nil {
-					fmt.Println("error in ReestablishConn: ", err1)
-				}
+	err = ExponentialBackoff(*defaultBackoff, func() (bool, error) {
+	start:
+		live, err = m.Client.GetInstanceIds(key, aurora.LIVE_STATES)
+		if err != nil {
+			err1 := m.Client.ReestablishConn()
+			if err1 != nil {
+				return false, nil
 			}
-
+			goto start
```

Comment:
This is actually a really clever way of handling the fact that it's a retryable error. But the ExponentialBackoff function should handle this just fine. A return false, nil indicates a retryable error.

Reply:
Yeah might be a bad idea to use goto cause it will be in an infinite loop is the connection fails.